src/two_player_collision_example.cpp

```diff
@@ -0,0 +1,357 @@
+static const Dimension kP2PhiIdx = P1::kNumXDims + P2::kPhiIdx;
+static const Dimension kP2VIdx = P1::kNumXDims + P2::kVIdx;
+static const Dimension kP2AIdx = P1::kNumXDims + P2::kAIdx;
+
+// Control dimensions.
+static const Dimension kP1OmegaIdx = 0;
+static const Dimension kP1JerkIdx = 1;
+static const Dimension kP2OmegaIdx = 0;
+static const Dimension kP2JerkIdx = 1;
+}  // anonymous namespace
+
+TwoPlayerCollisionExample::TwoPlayerCollisionExample(
+    const SolverParams& params) {
+  // Create dynamics.
+  const std::shared_ptr<const ConcatenatedDynamicalSystem> dynamics(
+      new ConcatenatedDynamicalSystem(
+          {std::make_shared<SinglePlayerCar6D>(kInterAxleLength),
+           std::make_shared<SinglePlayerCar6D>(kInterAxleLength)},
+          kTimeStep));
+
+  // Set up initial state.
+  x0_ = VectorXf::Zero(dynamics->XDim());
+  x0_(kP1XIdx) = kP1InitialX;
+  x0_(kP1YIdx) = kP1InitialY;
+  x0_(kP1HeadingIdx) = kP1InitialHeading;
+  x0_(kP1VIdx) = kP1InitialSpeed;
+  x0_(kP2XIdx) = kP2InitialX;
+  x0_(kP2YIdx) = kP2InitialY;
+  x0_(kP2HeadingIdx) = kP2InitialHeading;
+  x0_(kP2VIdx) = kP2InitialSpeed;
+
+  // Set up initial strategies and operating point.
+  strategies_.reset(new std::vector<Strategy>());
+  for (PlayerIndex ii = 0; ii < dynamics->NumPlayers(); ii++)
+    strategies_->emplace_back(kNumTimeSteps, dynamics->XDim(),
+                              dynamics->UDim(ii));
+
+  operating_point_.reset(
+      new OperatingPoint(kNumTimeSteps, dynamics->NumPlayers(), 0.0, dynamics));
+
+  // Set up costs for all players.
+  PlayerCost p1_cost, p2_cost;
+
+  // Orientation cost
+  const auto p1_nominal_orientation_cost = std::make_shared<OrientationCost>(
+      kNominalHeadingCostWeight, kP1HeadingIdx, kP1NominalHeading,
+      "NominalHeadingP1");
+  // p1_cost.AddStateCost(p1_nominal_orientation_cost);
+  const auto p2_nominal_orientation_cost = std::make_shared<OrientationCost>(
+      kNominalHeadingCostWeight, kP2HeadingIdx, kP1NominalHeading,
+      "NominalHeadingP2");
+  // p2_cost.AddStateCost(p2_nominal_orientation_cost);
+  
+  // cost for deviating from the center of the lane (for both p1 and p2)
+  const Polyline2 lane1_p1p2(
+      {Point2(2.5, -50.0), Point2(2.5, 50.0)});
+  const std::shared_ptr<QuadraticPolyline2Cost> p1_lane1_cost(
+      new QuadraticPolyline2Cost(kLaneCostWeight, lane1_p1p2, {kP1XIdx, kP1YIdx},
+                                 "LaneCenter"));
+  const std::shared_ptr<QuadraticPolyline2Cost> p2_lane1_cost(
+      new QuadraticPolyline2Cost(kLaneCostWeight * 10, lane1_p1p2, {kP2XIdx, kP2YIdx},
+                                 "LaneCenter"));
+  p1_cost.AddStateCost(p1_lane1_cost);
+  p2_cost.AddStateCost(p2_lane1_cost);
+
+  // cost for leaving the left boundary of the lane (for both p1 and p2)
+  const std::shared_ptr<SemiquadraticPolyline2Cost> p1_lane1_l_cost(
+      new SemiquadraticPolyline2Cost(kLaneBoundaryCostWeight * 1000, lane1_p1p2,
```

Comment:
nit: please use 1000.0 instead of just 1000. basically this just keeps the compiler from having to guess whether it's floating point or integer